MapboxVision/src/test/java/com/mapbox/vision/telemetry/TotalBytesCounterTest.kt

```diff
@@ -0,0 +1,291 @@
+                        currentTime = CURRENT_TIME
+                    )
+                }
+
+                val sequenceOfBytes = listOfBytesAndResults.map { it.first }
+                val expectedValue = listOfBytesAndResults.map { it.second }
+                lateinit var actualValue: List<Boolean>
+
+                When("Send track bytes in sequence of bytes $sequenceOfBytes") {
+
+                    actualValue = sequenceOfBytes.map { bytes ->
+                        totalBytesCounter10Min10kBytes.trackSentBytes(bytes)
+                    }
+                }
+
+                Then("It should be $expectedValue") {
+                    assertEquals(expectedValue, actualValue)
+                }
+            }
+        }
+
+        Scenario("Check trackSentBytes with several sessions") {
+
+            val mockedTime = mockk<Time>()
+
+            lateinit var totalBytesCounter10Min10kBytes: TotalBytesCounter
+
+            Given("TotalBytesCounter with 10 minutes, 10 KBytes and started session") {
+                totalBytesCounter10Min10kBytes = getTotalBytesCounterWithStartedSession(
+                    sessionLengthMillis = TEST_SESSION_LENGTH_MS,
+                    sessionMaxBytes = TEST_SESSION_MAX_BYTES,
+                    mockedTime = mockedTime,
+                    currentTime = CURRENT_TIME
+                )
+            }
+
+            val successfulSequenceOfBytes = listOf(1_000L, 5_000L, 4_000L)
+            val unsuccessfulSequenceOfBytes = listOf(1_000L, 1L, 2_000L)
+
+            Then("Successfully send sequence of bytes $successfulSequenceOfBytes") {
+                successfulSequenceOfBytes.forEach { bytes ->
+                    assertTrue(totalBytesCounter10Min10kBytes.trackSentBytes(bytes))
+                }
+            }
+
+            Then("Unsuccessfully send sequence of bytes $unsuccessfulSequenceOfBytes") {
+                unsuccessfulSequenceOfBytes.forEach { bytes ->
+                    assertFalse(totalBytesCounter10Min10kBytes.trackSentBytes(bytes))
+                }
+            }
+
+            Then("Move time to the next session") {
+                every { mockedTime.millis() }.returns(CURRENT_TIME + TEST_SESSION_LENGTH_MS)
+            }
+
+            Then("Successfully send sequence of bytes $successfulSequenceOfBytes") {
+                successfulSequenceOfBytes.forEach { bytes ->
+                    assertTrue(totalBytesCounter10Min10kBytes.trackSentBytes(bytes))
+                }
+            }
+        }
+    }
+})
+
+private fun getTotalBytesCounterWithStartedSession(
+    sessionLengthMillis: Long,
+    sessionMaxBytes: Long,
+    mockedTime: Time,
+    currentTime: Long
+): TotalBytesCounter {
+    val result = TotalBytesCounter.Impl(
+        sessionLengthMillis = sessionLengthMillis,
+        sessionMaxBytes = sessionMaxBytes,
+        time = mockedTime
+    )
+    every { mockedTime.millis() }.returns(currentTime)
+    assertEquals(result.millisToNextSession(), 0)
```

Comment:
Do we need an assertion here?

Reply:
I found one case:
when system.timeMillis return 0 as current time, some logic breaks down. just for checking that case.
now there is no fix for such situation.